### ISSUE TITLE
Why WebView don't lose previous cookie or session after first run

### ISSUE THREAD

**Issue by telmen** [facebook/react-native#13562]
### Description

After running app first time, I want to remove a previous session, cookie or something like these of WebView. I'm trying to authenticate an Instagram user with OAuth. When on the first run it works ok. After the first run, I want to let the user to login with another Instagram account or something else. But WebView already having some storage in it and directs to redirect URL immediately which is accessed the previous request. How do I lose state or storage, cookie or something of WebView and on every new component mount I want to erase previous records? Sorry, :badenglish: 

### Reproduction Steps and Sample Code

here is simple snack https://snack.expo.io/HJRKKTmAx 

  ***An example of your code is REQUIRED***

### Additional Information

* React Native version: [0.42.3]
* Platform: [both]
* Development Operating System: [Linux]
* Dev tools: [CRNA, Expo]

**Comment by israelidanny** [comment]
This is actually something I'm trying to do as well. Would love to know the answer.

**Comment by telmen** [comment]
Still no answer. 😞

**Comment by Gaafar** [comment]
Any updates on this issue? any way to clear webview cookies?

**Comment by Gaafar** [comment]
@telmen, @israelidanny I found a way to delete cookies using this lib https://github.com/joeferraro/react-native-cookies 
function: `CookieManager.clearAll`

**Comment by aseemk** [comment]
Running into this too and wishing there were a first-class way to at least clear all cookies (or equivalent of an incognito/ephemeral session). Will try the @joeferraro module, thanks!

**Comment by tobyf93** [comment]
Having the same issue :( i was hoping to find a static method on the WebView component but unfortunately no such thing...

**Comment by mightym** [comment]
Same issue here. Using Instagram OAuth to login the user to my app. Afterwards the user can't login via another account. Cookies will still login himself with the first account he used.

**Comment by mightym** [comment]
Found a simple unconventional way to solve this :) 

Just create a webview where you log out/log in the user and render a webview with instagrams logout url (gladly its a GET request :) ) 

`<WebView source={{ uri: 'https://instagram.com/accounts/logout/' }} style={{ height:0 }}/>`

**Comment by rastogitech** [comment]
It's not solved yet.!!

**Comment by jdnichollsc** [comment]
Same issue with other external authentication :/

**Comment by jdnichollsc** [comment]
@react-native-bot A question? This is a real issue!

**Comment by mightym** [comment]
@jdnichollsc right. I also think this is more a missing implementation than a question.

**Comment by Fanatic17** [comment]
I think we need some explicit cookie reset mechanisms. I couldn't find a simple way to do it

**Comment by jdnichollsc** [comment]
@sahrens please reopen this issue
Thanks in advance!

**Comment by sourabhlk** [comment]
Anyone solved this issue.

**Comment by marcio704** [comment]
Same problem here

**Comment by sixumeng** [comment]
Don't know if this could work, but can anyone try this?

` injectedJavaScript={`
        document.cookie
             .split(";")
                  .forEach(function(c) {
                 document.cookie = c.replace(/^ +/, "").replace(/=.*/, "=;expires=" + new Date().toUTCString() + ";domain=your domain");
             });
             window.postMessage();
    `}`

**Comment by feliperaul** [comment]
I just wished I knew about this bug before commiting so much code to WebView ...

**Comment by javierbustamante** [comment]
This issue must be reopened!! "react-native-cookies" library solve the problem but is not longer maintained:
"I no longer use this library and I'm looking for maintainer(s) to take full control over the development and release cycle."

**Comment by mightym** [comment]
On my issue 
https://github.com/facebook/react-native/issues/13562#issuecomment-354110511 there is actually no way to solve this. And its not react native related or anything else. Its just how browsers work. If this is a secure cookie you can't just access it or delete it. It gets set by instagram.com. and only instagram.com is allowed to delete/access it. 
Facebook has an api endpoint where you can logout a user. Which invalidates the session/cookie. So basically in Instagrams case it a missing implementation of them. And especially as the removed logout on GET via `https://instagram.com/accounts/logout/` there is no way to solve this. Your user will be logged in as long the cookie/session is valid.

**Comment by mehulnahar** [comment]
https://instagram.com/accounts/logout/ 

This is also not working.

**Comment by mightym** [comment]
@mehulnahar logout on get is deprecated

**Comment by mhdSid** [comment]
Use cookie manager to clear cookies. I came across the problem and solved it by cleaning cookies using React Native Cookies. If you want to use the library, make sure to fix the linking in header search paths

**Comment by Normence** [comment]
Still having this issue. 
React-Native: 0.59.0
React-Native-Webview: 5.11.0

**Comment by jdnichollsc** [comment]
or moving React Native Cookies plugin to the https://github.com/react-native-community?

**Comment by ceribbo** [comment]
Same issue here!

**Comment by mhdSid** [comment]
Are you guys trying to clear cookies in a webview? use react-native-cookies

If it is a safariViewController then react-native-cookies doesn't solve the problem

**Comment by hueyAtFetchly** [comment]
This IS an issue, the component has no way to totally clear session, cache, cookies. This is required functionality from a WebView component. 

OR there should be incognito mode on WebView as with all major browsers. 

Clearing cookies is currently a solution for iOS, but does not work to the effect needed on Android.

Please update with a method for destroying session, or for incognito style mode.

```
import CookieManager from 'react-native-cookies';
CookieManager.clearAll();
```

**Comment by mhdSid** [comment]
Facebook tells you that any cookie based authentication is wrong and may cause bugs...

**Comment by Dhirajbhujbal** [comment]
Facing the same issue.

Have resolved this issue, but I imported the 'WebView' from react-native which is not recommended (webView has to be imported from react-native-webview)

```
import { View,  StyleSheet, Alert, NavState, WebView} from 'react-native';
const RCTNetworking = require('RCTNetworking');

componentDidMount() {
 RCTNetworking.clearCookies(() => true);
}
```

This has been worked successfully and I am able to see the login screen again.

My question is when I tried to import webview from 'react-native-webview' (react native recomended)
and clearing the cookies by using react-native-cookies (https://github.com/joeferraro/react-native-cookies)
it seems failed able to navigate directly on the dashboard without asking the login again.

```
import WebView from 'react-native-webview';
import CookieManager 'react-native-cookies'

componentDidMount() {
   CookieManager.clearAll().then( () =>  console.log(`All Cookies cleared`) ).
}
```

Note: Console log is printed successfully, this seems like clearAll has worked but It still navigate me into the app without asking the login again.

**Comment by mhdSid** [comment]
How about using async storage?!?!? 

In react native docs, cookie based authentication is not recommended. 

After suffering with cookie auth, I have just finalized a slick login system using async storage and that is the best way to go. 

If you’re having time to make this solution work and even posting on GitHub haven't solved your issue, then know that the path itself might be wrong.

**Comment by tkow** [comment]
It's going to  abandon WebView Authentication like OAuth until the official endpoints ready, if webview session is not accepted. Our application uses github and twitter authentication via webview and it's only way to use them , so I want to solve this.

**Comment by minmingsheng** [comment]
does `incognito` props help?
